py-glaredb/src/environment.rs

```diff
@@ -0,0 +1,67 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use datafusion::datasource::{DefaultTableSource, MemTable};
+use datafusion::error::Result as DataFusionResult;
+use datafusion::logical_expr::TableProviderFilterPushDown;
+use datafusion::{
+    arrow::{
+        datatypes::SchemaRef,
+        pyarrow::{PyArrowConvert, PyArrowType},
+        record_batch::RecordBatch,
+    },
+    datasource::TableProvider,
+    execution::context::SessionState,
+    logical_expr::{TableSource, TableType},
+    physical_plan::ExecutionPlan,
+    prelude::Expr,
+};
+use pyo3::{
+    exceptions::PyRuntimeError,
+    prelude::*,
+    types::{PyTuple, PyType},
+};
+use sqlexec::{
+    engine::{Engine, TrackedSession},
+    environment::EnvironmentReader,
+    parser,
+    session::ExecutionResult,
+};
+use std::any::Any;
+use std::sync::Arc;
+
+/// Read polars dataframes from the python environment.
+#[derive(Debug, Clone, Copy)]
+pub struct PolarsReader;
+
+impl EnvironmentReader for PolarsReader {
+    fn resolve_table(
+        &self,
+        name: &str,
+    ) -> Result<Option<Arc<dyn TableProvider>>, Box<dyn std::error::Error + Send + Sync>> {
+        Python::with_gil(|py| {
+            let var = py.eval(name, None, None).unwrap();
+            let polars_type: &PyType = py
+                .import("polars.dataframe.frame")
+                .unwrap()
+                .getattr("DataFrame")
```

Comment:
i _think_ this would work as well
```suggestion
                .import("polars")
                .unwrap()
                .getattr("DataFrame")
```